src/NuGet.Clients/NuGet.Credentials/SecurePluginCredentialProvider.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Configuration;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Plugins;
+using System.Linq;
+
+namespace NuGet.Credentials
+{
+    public class SecurePluginCredentialProvider : ICredentialProvider
```

Comment:
Either add XML doc comments to the type and all public members or don't any at all.  Two members have XML doc comments already.

Also, can be `sealed`?